core/src/main/scala/org/apache/spark/scheduler/ExecutorDecommissionInfo.scala

```diff
@@ -25,7 +25,19 @@ package org.apache.spark.scheduler
  *                shuffle data might be lost even if the external shuffle service is enabled.
  */
 private[spark]
-case class ExecutorDecommissionInfo(message: String, workerHost: Option[String] = None)
+abstract class DecommissionInfo
```

Comment:
If all subclasses must to be defined in this file, maybe `sealed`?

Reply:
OK, sounds good!